### ISSUE TITLE
wr.redshift.copy() keeps files in s3 when exception is raised

### ISSUE THREAD

**Issue by njdanielsen** [awslabs/aws-data-wrangler#505]
**Describe the bug**

When an error is thrown in the wr.redshift.copy() method, s3 files are not cleaned up and remain there.

**To Reproduce**
Steps to reproduce the behavior. Also add details about Python and Wrangler's version and how the library was installed.

I encountered this when I didn't specify a varchar length for the dataframe and the column exceeded the default for varchar. My expectation is that this method would respect the param for `keep_files` to be false, and would dispose of any files when in success or failure. 

```
wr.redshift.copy(
    df=df,
    path=path,
    con=con_redshift,
    schema="public",
    table="sample_data_table",
    mode="overwrite",
#     varchar_lengths={'data': 6500}
)
```

```
---------------------------------------------------------------------------
ProgrammingError                          Traceback (most recent call last)
<ipython-input-39-489fc07fb9d7> in <module>
      2 
      3 
----> 4 wr.redshift.copy(
      5     df=df,
      6     path=path,

~/miniconda3/envs/leadgen/lib/python3.8/site-packages/awswrangler/redshift.py in copy(df, path, con, table, schema, iam_role, aws_access_key_id, aws_secret_access_key, aws_session_token, index, dtype, mode, diststyle, distkey, sortstyle, sortkey, primary_keys, varchar_lengths_default, varchar_lengths, keep_files, use_threads, boto3_session, s3_additional_kwargs, max_rows_by_file)
   1345         max_rows_by_file=max_rows_by_file,
   1346     )
-> 1347     copy_from_files(
   1348         path=path,
   1349         con=con,

~/miniconda3/envs/leadgen/lib/python3.8/site-packages/awswrangler/redshift.py in copy_from_files(path, con, table, schema, iam_role, aws_access_key_id, aws_secret_access_key, aws_session_token, parquet_infer_sampling, mode, diststyle, distkey, sortstyle, sortkey, primary_keys, varchar_lengths_default, varchar_lengths, path_suffix, path_ignore_suffix, use_threads, boto3_session, s3_additional_kwargs)
   1170                 s3_additional_kwargs=s3_additional_kwargs,
   1171             )
-> 1172             _copy(
   1173                 cursor=cursor,
   1174                 path=path,

~/miniconda3/envs/leadgen/lib/python3.8/site-packages/awswrangler/redshift.py in _copy(cursor, path, table, iam_role, aws_access_key_id, aws_secret_access_key, aws_session_token, boto3_session, schema)
    112     sql: str = f"COPY {table_name} FROM '{path}'{auth_str}\nFORMAT AS PARQUET"
    113     _logger.debug("copy query:\n%s", sql)
--> 114     cursor.execute(sql)
    115 
    116 

~/miniconda3/envs/leadgen/lib/python3.8/site-packages/redshift_connector/cursor.py in execute(self, operation, args, stream, merge_socket_read)
    160                 self._c.execute(self, "begin transaction", None)
    161             self._c.merge_socket_read = merge_socket_read
--> 162             self._c.execute(self, operation, args)
    163         except AttributeError as e:
    164             raise e

~/miniconda3/envs/leadgen/lib/python3.8/site-packages/redshift_connector/core.py in execute(self, cursor, operation, vals)
   1095             self.handle_messages_merge_socket_read(cursor)
   1096         else:
-> 1097             self.handle_messages(cursor)
   1098 
   1099     def _send_message(self: "Connection", code: bytes, data: bytes) -> None:

~/miniconda3/envs/leadgen/lib/python3.8/site-packages/redshift_connector/core.py in handle_messages(self, cursor)
   1170 
   1171         if self.error is not None:
-> 1172             raise self.error
   1173 
   1174     def handle_messages_merge_socket_read(self: "Connection", cursor: Cursor):

ProgrammingError: {'S': 'ERROR', 'C': 'XX000', 'M': 'Spectrum Scan Error', 'D': '\n  -----------------------------------------------\n  error:  Spectrum Scan Error\n  code:      15001\n  context:   The length of the data column data is longer than the length defined in the table. Table: 256, Data: 2737, File name: https://s3.us-west-2.amazonaws.com/{**}/3a13c036d1ac_0.snappy.parquet?&amp;X-Amz-Al\n  query:     69281625\n  location:  dory_util.cpp:945\n  process:   fetchtask_thread [pid=1668]\n  -----------------------------------------------\n', 'F': '/home/ec2-user/padb/src/sys/xen_execute.cpp', 'L': '8948', 'R': 'pg_throw'}
```

Outside of that, this library is so helpful! Thanks for the work on this.

**Comment by igorborgest** [comment]
@njdanielsen Thank you for reporting it!

Released on version [2.3.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/2.3.0) 🚀

**Comment by njdanielsen** [comment]
@igorborgest solid work on this library.
